<launch>
    <!-- Config map file -->
    <arg name="world" default="raicom" />   <!--  tianracer_racetrack  test_indoor  racetrack_1   raicom -->
    <arg name="robot_name" default="$(optenv TIANRACER_NAME /)" doc="robot name [tianracer_No1, tianracer_No2, tianracer_No3, ...]"/>
    <arg name="namespace" default="tianracer"/>

    <!-- rviz -->
    <group if="$(eval arg('robot_name') == '/')">

        <!-- Map server, load map-->
        <node name="map_server" pkg="map_server" type="map_server" ns="$(arg namespace)" args="$(find tianracer_gazebo)/maps/$(arg world).yaml" >
            <param name="frame_id" value="map" />
        </node>

        <!-- launch gazebo simulation environment -->
        <include file="$(find tianracer_gazebo)/launch/tianracer_on_racetrack.launch">
            <arg name="world" value="$(arg world)" />
            <arg name="x_pos" default="0.1" />
            <arg name="y_pos" default="0" />
            <arg name="z_pos" default="1" />
            <arg name="R_pos" default="0" />
            <arg name="P_pos" default="0" />
            <arg name="Y_pos" default="1.54" />
            <arg name="robot_name" value="$(arg namespace)" />
        </include>

        <!-- launch tianracer control -->
        <include file="$(find tianracer_gazebo)/launch/tianracer_control.launch" >
            <arg name="robot_name" value="$(arg namespace)" />
        </include>

        <!-- move base -->
        <include file="$(find tianracer_gazebo)/launch/includes/teb_base.launch.xml" >
            <arg name="robot_name" value="$(arg namespace)" />
        </include>

        <!-- AMCL -->
        <include file="$(find tianracer_gazebo)/launch/includes/amcl.launch.xml" >
            <arg name="initial_pose_x" default="0.1"/>
            <arg name="initial_pose_y" default="0.0"/>
            <arg name="initial_pose_a" default="1.54"/>
            <arg name="robot_name" value="$(arg namespace)" />
        </include>
        <!-- path plan 
        <node pkg="tianracer_navigation" type="plan" name="plan" output="screen"/>  -->

        <!-- rf2o_laser_odometry-->
        <include file="$(find rf2o_laser_odometry)/launch/rf2o_laser_odometry.launch" >
        </include>
        <include file="$(find robot_localization)/launch/ukf_template.launch" >
        </include>       

        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tianracer_gazebo)/rviz/nav.rviz"/>
    </group>

    <group unless="$(eval arg('robot_name') == '/')">

        <!-- Map server, load map-->
        <node name="map_server" pkg="map_server" type="map_server" ns="$(arg robot_name)" args="$(find tianracer_gazebo)/maps/$(arg world).yaml" >
            <param name="frame_id" value="map" />
        </node>

        <!-- launch gazebo simulation environment -->
        <include file="$(find tianracer_gazebo)/launch/tianracer_on_racetrack.launch">
            <arg name="world" value="$(arg world)" />
            <arg name="robot_name" value="$(arg robot_name)" />
        </include>

        <!-- spawn model-->
        <!-- <include file="$(find tianracer_gazebo)/launch/spawn_model.launch" /> -->

        <!-- launch tianracer control -->
        <include file="$(find tianracer_gazebo)/launch/tianracer_control.launch" >
            <arg name="robot_name" value="$(arg robot_name)" />
        </include>

        <!-- move base -->
        <include file="$(find tianracer_gazebo)/launch/includes/teb_base.launch.xml" >
            <arg name="robot_name" value="$(arg robot_name)" />
        </include>

        <!-- AMCL -->
        <include file="$(find tianracer_gazebo)/launch/includes/amcl.launch.xml" >
            <arg name="robot_name" value="$(arg robot_name)" />
        </include>

        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tianracer_gazebo)/rviz/nav_$(arg robot_name).rviz"/>
    </group>

</launch>
